File: decompiled/3.18.0/be/ehealth/businessconnector/ehbox/v3/builders/impl/ConsultationFullMessageBuilder.java
package be.ehealth.businessconnector.ehbox.v3.builders.impl;

import be.ehealth.businessconnector.ehbox.api.domain.AcknowledgeMessage;
import be.ehealth.businessconnector.ehbox.api.domain.Addressee;
import be.ehealth.businessconnector.ehbox.api.domain.Document;
import be.ehealth.businessconnector.ehbox.api.domain.DocumentMessage;
import be.ehealth.businessconnector.ehbox.api.domain.ErrorMessage;
import be.ehealth.businessconnector.ehbox.api.domain.Message;
import be.ehealth.businessconnector.ehbox.api.domain.exception.EhboxBusinessConnectorException;
import be.ehealth.technicalconnector.enumeration.Charset;
import be.ehealth.technicalconnector.exception.TechnicalConnectorException;
import be.ehealth.technicalconnector.exception.TechnicalConnectorExceptionValues;
import be.ehealth.technicalconnector.exception.UnsealConnectorException;
import be.ehealth.technicalconnector.service.etee.Crypto;
import be.ehealth.technicalconnector.utils.ConnectorIOUtils;
import be.ehealth.technicalconnector.utils.IdentifierType;
import be.fgov.ehealth.ehbox.consultation.protocol.v3.ConsultationAnnexType;
import be.fgov.ehealth.ehbox.consultation.protocol.v3.ConsultationContentType;
import be.fgov.ehealth.ehbox.consultation.protocol.v3.ConsultationDocumentType;
import be.fgov.ehealth.ehbox.consultation.protocol.v3.ConsultationMessageType;
import be.fgov.ehealth.ehbox.consultation.protocol.v3.ContentContextType;
import be.fgov.ehealth.ehbox.consultation.protocol.v3.DestinationContextType;
import be.fgov.ehealth.ehbox.consultation.protocol.v3.GetFullMessageResponse;
import be.fgov.ehealth.ehbox.core.v3.ContentSpecificationType;
import be.fgov.ehealth.ehbox.core.v3.ErrorType;
import be.fgov.ehealth.ehbox.core.v3.FreeInformationsType;
import be.fgov.ehealth.ehbox.core.v3.Row;
import be.fgov.ehealth.ehbox.core.v3.Table;
import java.io.IOException;
import java.io.InputStream;
import java.util.Iterator;
import java.util.List;
import javax.activation.DataHandler;
import org.apache.commons.lang.ArrayUtils;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class ConsultationFullMessageBuilder extends AbstractConsultationBuilder<GetFullMessageResponse> {
   private static final Logger LOG = LoggerFactory.getLogger(ConsultationFullMessageBuilder.class);

   /** @deprecated */
   @Deprecated
   public ConsultationFullMessageBuilder(Crypto crypto) {
      this();
   }

   public ConsultationFullMessageBuilder() {
   }

   public final Message<GetFullMessageResponse> buildFullMessage(GetFullMessageResponse response) throws EhboxBusinessConnectorException, TechnicalConnectorException {
      ConsultationMessageType recievedMsg = response.getMessage();
      Message<GetFullMessageResponse> message = this.createMessage(recievedMsg.getContentContext().getContentSpecification(), response, recievedMsg.getMessageId(), response.getMessage().getPublicationId());
      AbstractConsultationBuilder.ExceptionContainer<GetFullMessageResponse> container = new AbstractConsultationBuilder.ExceptionContainer(message);
      this.processMessageInfo(response.getMessageInfo(), message);
      this.processSender(response.getSender(), (ContentSpecificationType)null, message);
      this.processDestinationContext(recievedMsg.getDestinationContexts(), message);
      this.processContentContext(recievedMsg.getContentContext(), message, container);
      return container.getMessage();
   }

   private void processContentContext(ContentContextType context, Message<GetFullMessageResponse> message, AbstractConsultationBuilder.ExceptionContainer<GetFullMessageResponse> container) throws TechnicalConnectorException, EhboxBusinessConnectorException {
      this.processContentSpecification(context.getContentSpecification(), message);
      this.processCustomMetas(context.getCustomMetas(), message);
      if (message instanceof DocumentMessage) {
         this.processDocument(context.getContent(), message, container);
      } else if (message instanceof AcknowledgeMessage) {
         this.processAcknowledge(context.getContent(), message);
      } else if (message instanceof ErrorMessage) {
         this.processError(context.getContent(), message);
      }

   }

   private void processAcknowledge(ConsultationContentType response, Message<?> message) throws EhboxBusinessConnectorException, TechnicalConnectorException {
      LOG.debug("processAcknowledge : processing acknowledge for message " + message.getId() + " and response acknowledgement" + response.getAcknowledgment() + " : no special processing needed");
   }

   private void processError(ConsultationContentType response, Message<?> message) throws EhboxBusinessConnectorException, TechnicalConnectorException {
      if (message instanceof ErrorMessage) {
         ErrorMessage<?> errorMessage = (ErrorMessage)message;
         ErrorType error = response.getError();
         if (error != null) {
            errorMessage.setErrorCode(error.getCode());
            errorMessage.setErrorPublicationId(error.getPublicationId());
            Iterator i$;
            String failureMessage;
            if (error.getMessages() != null) {
               i$ = error.getMessages().iterator();

               while(i$.hasNext()) {
                  failureMessage = (String)i$.next();
                  errorMessage.getErrorMsg().add("error:" + failureMessage);
               }
            }

            if (error.getFailures() != null) {
               i$ = error.getFailures().iterator();

               while(i$.hasNext()) {
                  failureMessage = (String)i$.next();
                  errorMessage.getErrorMsg().add("failure:" + failureMessage);
               }
            }
         }
      }

   }

   private void processDocument(ConsultationContentType response, Message<?> message, AbstractConsultationBuilder.ExceptionContainer<GetFullMessageResponse> container) throws TechnicalConnectorException, EhboxBusinessConnectorException {
      DocumentMessage<?> documentMessage = (DocumentMessage)message;
      this.processINSSPatient(response.getEncryptableINSSPatient(), documentMessage, container);
      this.processFreeText(response.getFreeInformations(), documentMessage, container);
      this.processMainDocument(response.getDocument(), documentMessage, container);
      if (!response.getAnnices().isEmpty()) {
         documentMessage.setHasAnnex(true);
         Iterator i$ = response.getAnnices().iterator();

         while(i$.hasNext()) {
            ConsultationAnnexType annexType = (ConsultationAnnexType)i$.next();
            this.processAnnex(documentMessage, annexType, container);
         }
      }

   }

   private void processDestinationContext(List<DestinationContextType> destinationList, Message<?> message) {
      Iterator i$ = destinationList.iterator();

      while(i$.hasNext()) {
         DestinationContextType destinationContext = (DestinationContextType)i$.next();
         IdentifierType identityType = IdentifierType.lookup(destinationContext.getType(), destinationContext.getSubType(), 49);
         Addressee destination = new Addressee(identityType);
         destination.setId(destinationContext.getId());
         if (destinationContext.getUser() != null) {
            destination.setFirstName(destinationContext.getUser().getFirstName());
            destination.setLastName(destinationContext.getUser().getLastName());
         }

         destination.setQuality(destinationContext.getQuality());
         message.getDestinations().add(destination);
      }

   }

   private void processAnnex(DocumentMessage<?> documentMessage, ConsultationAnnexType annexType, AbstractConsultationBuilder.ExceptionContainer<GetFullMessageResponse> container) throws TechnicalConnectorException, EhboxBusinessConnectorException {
      Document annex = new Document();
      annex.setFilename(annexType.getDownloadFileName());
      annex.setMimeType(annexType.getMimeType());
      byte[] annexTitle = this.handleAndDecryptIfNeeded(annexType.getEncryptableTitle(), documentMessage.isEncrypted(), container);
      if (annexTitle != null) {
         annex.setTitle(ConnectorIOUtils.toString(annexTitle, Charset.UTF_8));
      }

      DataHandler annexHandler = null;
      if (annexType.getEncryptableBinaryContent() != null) {
         annexHandler = annexType.getEncryptableBinaryContent();
         if (annexHandler != null) {
            annex.setContent(this.base64decoding(annexHandler, documentMessage.isEncrypted(), container));
         }
      } else if (annexType.getEncryptableTextContent() != null) {
         annex.setContent(this.handleAndDecryptIfNeeded(annexType.getEncryptableTextContent(), documentMessage.isEncrypted(), container));
      }

      documentMessage.getAnnexList().add(annex);
   }

   private void processMainDocument(ConsultationDocumentType response, DocumentMessage<?> documentMessage, AbstractConsultationBuilder.ExceptionContainer<GetFullMessageResponse> container) throws TechnicalConnectorException, EhboxBusinessConnectorException {
      Document document = new Document();
      document.setFilename(response.getDownloadFileName());
      document.setMimeType(response.getMimeType());
      document.setTitle(response.getTitle());
      DataHandler responseHandler = null;
      if (response.getEncryptableBinaryContent() != null) {
         responseHandler = response.getEncryptableBinaryContent();
         if (responseHandler != null) {
            try {
               document.setContent(this.base64decoding(responseHandler, documentMessage.isEncrypted(), container));
            } catch (UnsealConnectorException var7) {
               document.setException(var7);
            }
         }
      } else if (response.getEncryptableTextContent() != null) {
         document.setContent(this.handleAndDecryptIfNeeded(response.getEncryptableTextContent(), documentMessage.isEncrypted(), container));
      }

      documentMessage.setDocument(document);
   }

   private void processINSSPatient(byte[] encryptableINSSPatient, DocumentMessage<?> documentMessage, AbstractConsultationBuilder.ExceptionContainer<GetFullMessageResponse> container) throws TechnicalConnectorException, EhboxBusinessConnectorException {
      if (encryptableINSSPatient != null) {
         boolean encrypted = documentMessage.isEncrypted();
         byte[] patientInss = this.handleAndDecryptIfNeeded(encryptableINSSPatient, encrypted, container);
         if (patientInss != null) {
            documentMessage.setPatientInss(ConnectorIOUtils.toString(patientInss, Charset.UTF_8));
            documentMessage.setEncrypted(encrypted);
         }
      }

   }

   private <T> void processFreeText(FreeInformationsType response, DocumentMessage<?> documentMessage, AbstractConsultationBuilder.ExceptionContainer<GetFullMessageResponse> container) throws TechnicalConnectorException, EhboxBusinessConnectorException {
      if (response != null) {
         byte[] freeText = this.handleAndDecryptIfNeeded(response.getEncryptableFreeText(), documentMessage.isEncrypted(), container);
         if (freeText != null && freeText.length > 0) {
            documentMessage.setFreeText(ConnectorIOUtils.toString(freeText, Charset.UTF_8));
         }

         Table table = response.getTable();
         if (table != null) {
            documentMessage.setFreeInformationTableTitle(table.getTitle());
            Iterator i$ = table.getRows().iterator();

            while(i$.hasNext()) {
               Row row = (Row)i$.next();
               byte[] decryptedKeyBytes = this.handleAndDecryptIfNeeded(row.getEncryptableLeftCell(), documentMessage.isEncrypted(), container);
               byte[] decryptedValueBytes = this.handleAndDecryptIfNeeded(row.getEncryptableRightCell(), documentMessage.isEncrypted(), container);
               String decryptedKey = ConnectorIOUtils.toString(decryptedKeyBytes, Charset.UTF_8);
               String decryptedValue = ConnectorIOUtils.toString(decryptedValueBytes, Charset.UTF_8);
               documentMessage.getFreeInformationTableRows().put(decryptedKey, decryptedValue);
            }
         }
      }

   }

   private <T> byte[] base64decoding(DataHandler dataHandler, boolean encrypted, AbstractConsultationBuilder.ExceptionContainer<GetFullMessageResponse> container) throws TechnicalConnectorException, EhboxBusinessConnectorException {
      InputStream is = null;

      byte[] var6;
      try {
         is = dataHandler.getInputStream();
         byte[] byteVal = ConnectorIOUtils.getBytes(is);
         if (ArrayUtils.isEmpty(byteVal)) {
            Object var12 = null;
            return (byte[])var12;
         }

         var6 = this.handleAndDecryptIfNeeded(byteVal, encrypted, container);
      } catch (IOException var10) {
         throw new TechnicalConnectorException(TechnicalConnectorExceptionValues.ERROR_GENERAL, var10, new Object[]{"Unable to decode datahandler"});
      } finally {
         ConnectorIOUtils.closeQuietly((Object)is);
      }

      return var6;
   }
}

File: src/main/kotlin/org/taktik/connector/business/ehbox/v3/builders/impl/ConsultationFullMessageBuilder.kt
/*
 *
 * Copyright (C) 2018 Taktik SA
 *
 * This file is part of FreeHealthConnector.
 *
 * FreeHealthConnector is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation.
 *
 * FreeHealthConnector is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with FreeHealthConnector.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

package org.taktik.connector.business.ehbox.v3.builders.impl

import be.fgov.ehealth.ehbox.consultation.protocol.v3.ConsultationAnnexType
import be.fgov.ehealth.ehbox.consultation.protocol.v3.ConsultationContentType
import be.fgov.ehealth.ehbox.consultation.protocol.v3.ConsultationDocumentType
import be.fgov.ehealth.ehbox.consultation.protocol.v3.ContentContextType
import be.fgov.ehealth.ehbox.consultation.protocol.v3.DestinationContextType
import be.fgov.ehealth.ehbox.consultation.protocol.v3.GetFullMessageResponse
import be.fgov.ehealth.ehbox.core.v3.FreeInformationsType
import org.apache.commons.lang.ArrayUtils
import org.slf4j.LoggerFactory
import org.taktik.connector.business.ehbox.api.domain.*
import org.taktik.connector.business.ehbox.api.domain.exception.EhboxBusinessConnectorException
import org.taktik.connector.technical.enumeration.Charset
import org.taktik.connector.technical.exception.TechnicalConnectorException
import org.taktik.connector.technical.exception.TechnicalConnectorExceptionValues
import org.taktik.connector.technical.exception.UnsealConnectorException
import org.taktik.connector.technical.service.sts.security.impl.KeyStoreCredential
import org.taktik.connector.technical.utils.ConnectorIOUtils
import org.taktik.connector.technical.utils.IdentifierType

import javax.activation.DataHandler
import java.io.IOException
import java.io.InputStream

class ConsultationFullMessageBuilder : AbstractConsultationBuilder<GetFullMessageResponse>() {

    @Throws(EhboxBusinessConnectorException::class, TechnicalConnectorException::class)
    fun buildFullMessage(
        credential: KeyStoreCredential,
        response: GetFullMessageResponse
    ): Message<GetFullMessageResponse> {
        val receivedMsg = response.message
        val message =
            this.createMessage(
                receivedMsg.contentContext.contentSpecification,
                response,
                receivedMsg.messageId,
                response.message.publicationId
            )
        val container = AbstractConsultationBuilder.ExceptionContainer(message)
        this.processMessageInfo(response.messageInfo, message)
        this.processSender(response.sender, null, message)
        this.processDestinationContext(receivedMsg.destinationContexts, message)
        this.processContentContext(credential, receivedMsg.contentContext, message, container)
        return container.getMessage()
    }

    @Throws(TechnicalConnectorException::class, EhboxBusinessConnectorException::class)
    private fun processContentContext(
        credential: KeyStoreCredential,
        context: ContentContextType,
        message: Message<GetFullMessageResponse>,
        container: AbstractConsultationBuilder.ExceptionContainer<GetFullMessageResponse>
    ) {
        this.processContentSpecification(context.contentSpecification, message)
        this.processCustomMetas(context.customMetas, message)
        if (message is AcknowledgeMessage<*>) {
            this.processDocument(credential, context.content, message, container)
            this.processAcknowledge(context.content, message)
        } else if (message is DocumentMessage<*>) {
            this.processDocument(credential, context.content, message, container)
        } else if (message is ErrorMessage<*>) {
            this.processError(context.content, message)
        }
    }

    @Throws(EhboxBusinessConnectorException::class, TechnicalConnectorException::class)
    private fun processAcknowledge(response: ConsultationContentType, message: AcknowledgeMessage<*>) {
        message.acknowledgment = response.acknowledgment?.let { Acknowledgment(
            messageId = it.messageId,
            recipient = it.recipient?.let {
                Addressee(IdentifierType.lookup(it.type, it.subType, IdentifierType.EHBOX)).apply {
                    id = it.id
                    quality = it.quality
                    firstName = it.user?.firstName
                    lastName = it.user?.lastName
                }
            },
            ackType = it.ackType,
            dateTime = it.dateTime.millis
        ) }
    }

    @Throws(EhboxBusinessConnectorException::class, TechnicalConnectorException::class)
    private fun processError(response: ConsultationContentType, message: Message<*>) {
        if (message is ErrorMessage<*>) {
            val error = response.error
            if (error != null) {
                message.errorCode = error.code
                message.errorPublicationId = error.publicationId
                if (error.messages != null) {
                    for (failureMessage in error.messages) {
                        message.errorMsg.add("error:" + failureMessage)
                    }
                }

                if (error.failures != null) {
                    for (failureMessage in error.failures) {
                        message.errorMsg.add("failure:" + failureMessage)
                    }
                }
            }
        }
    }

    @Throws(TechnicalConnectorException::class, EhboxBusinessConnectorException::class)
    private fun processDocument(
        credential: KeyStoreCredential,
        response: ConsultationContentType,
        message: Message<*>,
        container: AbstractConsultationBuilder.ExceptionContainer<GetFullMessageResponse>
    ) {
        val documentMessage = message as DocumentMessage<*>
        this.processINSSPatient(credential, response.encryptableINSSPatient, documentMessage, container)
        this.processFreeText(credential, response.freeInformations, documentMessage, container)
        this.processMainDocument(credential, response.document, documentMessage, container)
        if (!response.annices.isEmpty()) {
            documentMessage.isHasAnnex = true

            for (annexType in response.annices) {
                this.processAnnex(credential, documentMessage, annexType, container)
            }
        }
    }

    private fun processDestinationContext(destinationList: List<DestinationContextType>, message: Message<*>) {

        for (destinationContext in destinationList) {
            val identityType = IdentifierType.lookup(destinationContext.type, destinationContext.subType, 49)
            val destination = Addressee(identityType)
            destination.id = destinationContext.id
            if (destinationContext.user != null) {
                destination.firstName = destinationContext.user.firstName
                destination.lastName = destinationContext.user.lastName
            }

            destination.quality = destinationContext.quality
            message.getDestinations().add(destination)
        }
    }

    @Throws(TechnicalConnectorException::class, EhboxBusinessConnectorException::class)
    private fun processAnnex(
        credential: KeyStoreCredential,
        documentMessage: DocumentMessage<*>,
        annexType: ConsultationAnnexType,
        container: AbstractConsultationBuilder.ExceptionContainer<GetFullMessageResponse>
    ) {
        val annex = Document()
        annex.filename = annexType.downloadFileName
        annex.mimeType = annexType.mimeType
        val annexTitle = annexType.encryptableTitle?.let {
            this.handleAndDecryptIfNeeded(
                credential,
                it,
                documentMessage.isEncrypted,
                container
                                         )
        }
        if (annexTitle != null) {
            annex.title = ConnectorIOUtils.toString(annexTitle, Charset.UTF_8)
        }

        val annexHandler: DataHandler?
        if (annexType.encryptableBinaryContent != null) {
            annexHandler = annexType.encryptableBinaryContent
            if (annexHandler != null) {
                annex.setContent(
                    this.base64decoding(
                        credential,
                        annexHandler,
                        documentMessage.isEncrypted,
                        container
                    )
                )
            }
        } else if (annexType.encryptableTextContent != null) {
            annex.setContent(
                this.handleAndDecryptIfNeeded(
                    credential,
                    annexType.encryptableTextContent,
                    documentMessage.isEncrypted,
                    container
                )
            )
        }

        documentMessage.annexList.add(annex)
    }

    @Throws(TechnicalConnectorException::class, EhboxBusinessConnectorException::class)
    private fun processMainDocument(
        credential: KeyStoreCredential,
        response: ConsultationDocumentType,
        documentMessage: DocumentMessage<*>,
        container: AbstractConsultationBuilder.ExceptionContainer<GetFullMessageResponse>
    ) {
        val document = Document()
        document.filename = response.downloadFileName
        document.mimeType = response.mimeType
        document.title = response.title
        val responseHandler: DataHandler?
        if (response.encryptableBinaryContent != null) {
            responseHandler = response.encryptableBinaryContent
            if (responseHandler != null) {
                try {
                    document.setContent(
                        this.base64decoding(
                            credential,
                            responseHandler,
                            documentMessage.isEncrypted,
                            container
                        )
                    )
                } catch (var7: UnsealConnectorException) {
                    document.setException(var7)
                }
            }
        } else if (response.encryptableTextContent != null) {
            document.setContent(
                this.handleAndDecryptIfNeeded(
                    credential,
                    response.encryptableTextContent,
                    documentMessage.isEncrypted,
                    container
                )
            )
        }

        documentMessage.document = document
    }

    @Throws(TechnicalConnectorException::class, EhboxBusinessConnectorException::class)
    private fun processINSSPatient(
        credential: KeyStoreCredential,
        encryptableINSSPatient: ByteArray?,
        documentMessage: DocumentMessage<*>,
        container: AbstractConsultationBuilder.ExceptionContainer<GetFullMessageResponse>
    ) {
        if (encryptableINSSPatient != null) {
            val encrypted = documentMessage.isEncrypted
            val patientInss =
                this.handleAndDecryptIfNeeded(credential, encryptableINSSPatient, encrypted, container)
            if (patientInss != null) {
                documentMessage.patientInss = ConnectorIOUtils.toString(patientInss, Charset.UTF_8)
                documentMessage.isEncrypted = encrypted
            }
        }
    }

    @Throws(TechnicalConnectorException::class, EhboxBusinessConnectorException::class)
    private fun processFreeText(
        credential: KeyStoreCredential,
        response: FreeInformationsType?,
        documentMessage: DocumentMessage<*>,
        container: AbstractConsultationBuilder.ExceptionContainer<GetFullMessageResponse>
    ) {
        if (response != null) {
            val freeText =
                this.handleAndDecryptIfNeeded(
                    credential,
                    response.encryptableFreeText,
                    documentMessage.isEncrypted,
                    container
                )
            if (freeText != null && freeText.size > 0) {
                documentMessage.freeText = ConnectorIOUtils.toString(freeText, Charset.UTF_8)
            }

            val table = response.table
            if (table != null) {
                documentMessage.freeInformationTableTitle = table.title

                for (row in table.rows) {
                    val decryptedKeyBytes =
                        this.handleAndDecryptIfNeeded(
                            credential,
                            row.encryptableLeftCell,
                            documentMessage.isEncrypted,
                            container
                        )
                    val decryptedValueBytes =
                        this.handleAndDecryptIfNeeded(
                            credential,
                            row.encryptableRightCell,
                            documentMessage.isEncrypted,
                            container
                        )
                    val decryptedKey = ConnectorIOUtils.toString(decryptedKeyBytes, Charset.UTF_8)
                    val decryptedValue = ConnectorIOUtils.toString(decryptedValueBytes, Charset.UTF_8)
                    documentMessage.freeInformationTableRows.put(decryptedKey, decryptedValue)
                }
            }
        }
    }

    @Throws(TechnicalConnectorException::class, EhboxBusinessConnectorException::class)
    private fun base64decoding(
        credential: KeyStoreCredential,
        dataHandler: DataHandler,
        encrypted: Boolean,
        container: AbstractConsultationBuilder.ExceptionContainer<GetFullMessageResponse>
    ): ByteArray? {
        var inputStream: InputStream? = null

        val result: ByteArray?
        try {
            inputStream = dataHandler.inputStream
            val byteVal = ConnectorIOUtils.getBytes(inputStream)
            if (ArrayUtils.isEmpty(byteVal)) {
                return null
            }
            result = this.handleAndDecryptIfNeeded(credential, byteVal, encrypted, container)
        } catch (ex: IOException) {
            throw TechnicalConnectorException(
                TechnicalConnectorExceptionValues.ERROR_GENERAL,
                ex,
                "Unable to decode datahandler"
            )
        } finally {
            ConnectorIOUtils.closeQuietly(inputStream)
        }
        return result
    }

    companion object {
        private val LOG = LoggerFactory.getLogger(ConsultationFullMessageBuilder::class.java)
    }
}
